solr/core/src/test/org/apache/solr/cloud/AliasIntegrationTest.java

```diff
@@ -291,20 +316,25 @@ public void testModifyPropertiesCAR() throws Exception {
     setAliasProperty.addProperty("foo", "baz");
     setAliasProperty.addProperty("bar", "bam");
     setAliasProperty.process(cluster.getSolrClient());
-    checkFooAndBarMeta(aliasName, zkStateReader);
+    checkFooAndBarMeta(aliasName, zkStateReader, "baz", "bam");
 
     // now verify we can delete
     setAliasProperty = CollectionAdminRequest.setAliasProperty(aliasName);
     setAliasProperty.addProperty("foo", "");
     setAliasProperty.process(cluster.getSolrClient());
+    checkFooAndBarMeta(aliasName, zkStateReader, null, "bam");
+
+    // TODO is this a bug? should "bar" be null after the following
```

Comment:
@gerlowskija this might be a bug? I added some assertions to this older test and results are a bit surprising. my understanding is that setting value to null should remove it, but it doesn't. maybe I am reading it wrong.

Reply:
I'm not sure whether it's a bug or not, but I agree that it's very surprising.  It seems like empty-string is the only way to delete alias properties.  Might be worth a thread on dev@ to see whether anyone knows if this is intentional or not.

Reply:
email sent to dev list